[core] Can't install `@mui/material@next` and `@mui/material-nextjs@next`

### Steps to reproduce

Steps:
1. Run 
    ```
    mkdir foo && cd foo && npm init --yes && npm install @mui/material@next @mui/material-nextjs@next
    ```



### Current behavior

Error:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: foo@1.0.0
npm ERR! Found: @mui/material@6.0.0-beta.2
npm ERR! node_modules/@mui/material
npm ERR!   @mui/material@"6.0.0-beta.2" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @mui/material@"6.0.0-beta.1" from @mui/material-nextjs@6.0.0-alpha.14
npm ERR! node_modules/@mui/material-nextjs
npm ERR!   @mui/material-nextjs@"6.0.0-alpha.14" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```


### Expected behavior

No error

### Context

Could it be that `@mui/material-nextjs` doesn't always get published? It seems to break the peerDependency relationship.

This makes Toolpad core CLI fail. The problem should go away once v6 hits stable, but this may be something that needs to be fixed in our release process?

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>


**Search keywords**: nextjs peerdependencies

--- a/packages/mui-material-nextjs/package.json
+++ b/packages/mui-material-nextjs/package.json
@@ -50,7 +50,6 @@
     "@emotion/cache": "^11.11.0",
     "@emotion/react": "^11.11.4",
     "@emotion/server": "^11.11.0",
-    "@mui/material": "workspace:*",
     "@types/react": "^17.0.0 || ^18.0.0",
     "next": "^13.0.0 || ^14.0.0",
     "react": "^17.0.0 || ^18.0.0"

